core/src/main/java/org/apache/carbondata/core/writer/CarbonIndexFileMergeWriter.java

```diff
@@ -173,7 +173,7 @@ private String writeMergeIndexFileBasedOnSegmentFile(String segmentId,
           location =
               segmentFileStore.getTablePath() + CarbonCommonConstants.FILE_SEPARATOR + location;
         }
-        if (new Path(entry.getKey()).equals(new Path(location))) {
+        if (FileFactory.getCarbonFile(entry.getKey()).equals(FileFactory.getCarbonFile(location))) {
```

Comment:
Since this is inside a double for loop, we should avoid creating too many objects. Please describe what is missing in current approach. Is it because `//` is not replaced?